pkg/controllers/nodeclass/hash/suite_test.go

```diff
@@ -123,6 +142,73 @@ var _ = Describe("NodeClass Hash Controller", func() {
 		Entry("MetadataOptions Drift", &v1.EC2NodeClass{Spec: v1.EC2NodeClassSpec{MetadataOptions: &v1.MetadataOptions{HTTPEndpoint: aws.String("disabled")}}}),
 		Entry("Context Drift", &v1.EC2NodeClass{Spec: v1.EC2NodeClassSpec{Context: aws.String("context-2")}}),
 	)
+	It("should update nodeClaim annotation kubelet hash if nodePool was configured using v1beta1 NodePool ", func() {
+		kubeletConfig := &v1beta1.KubeletConfiguration{
+			ClusterDNS:                  []string{"test-cluster-dns"},
+			MaxPods:                     lo.ToPtr(int32(9383)),
+			PodsPerCore:                 lo.ToPtr(int32(9334283)),
+			SystemReserved:              map[string]string{"system-key": "reserved"},
+			KubeReserved:                map[string]string{"kube-key": "reserved"},
+			EvictionHard:                map[string]string{"eviction-key": "eviction"},
+			EvictionSoft:                map[string]string{"eviction-key": "eviction"},
+			EvictionSoftGracePeriod:     map[string]metav1.Duration{"test-soft-grace": {Duration: time.Hour}},
+			EvictionMaxPodGracePeriod:   lo.ToPtr(int32(382902)),
+			ImageGCHighThresholdPercent: lo.ToPtr(int32(382902)),
+			CPUCFSQuota:                 lo.ToPtr(false),
+		}
+		kubeletConfigString, _ := json.Marshal(kubeletConfig)
+		nodePool.Annotations = lo.Assign(nodePool.Annotations, map[string]string{
+			karpv1.KubeletCompatabilityAnnotationKey: string(kubeletConfigString),
+		})
+		nodeClaim := coretest.NodeClaim(karpv1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{karpv1.NodePoolLabelKey: nodePool.Name},
+				Annotations: map[string]string{
+					v1.AnnotationEC2NodeClassHash:        "123456",
+					v1.AnnotationEC2NodeClassHashVersion: "test",
+					v1.AnnotationKubeletHash:             "123456",
+				},
+			},
+			Spec: karpv1.NodeClaimSpec{
+				NodeClassRef: &karpv1.NodeClassReference{
+					Group: object.GVK(nodeClass).Group,
+					Kind:  object.GVK(nodeClass).Kind,
+					Name:  nodeClass.Name,
+				},
+			},
+		})
+		ExpectApplied(ctx, env.Client, nodeClass, nodeClaim, nodePool)
+		hashBefore := nodeClaim.Annotations[v1.AnnotationKubeletHash]
+
+		ExpectObjectReconciled(ctx, env.Client, hashController, nodeClass)
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.Annotations[v1.AnnotationKubeletHash]).ToNot(Equal(hashBefore))
```

Comment:
I think it might be worth us checking if the hash is update to an expectedHash value